src/Microsoft.ComponentDetection.Detectors/rust/SemVer/Comparator.cs

```diff
@@ -120,6 +116,10 @@ public enum Operator
             GreaterThanOrEqual,
         }
 
+        public Operator ComparatorType { get; set; }
+
+        public SemVersion Version { get; set; }
+
```

Comment:
`readonly`?

Reply:
it was necessary to convert field into a property to resolve [SA1401 : field should be private](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1401.md). And read-only is not valid on a property. 

However, we should not be providing setter for this property and I fixed that by removing the setter.